UPGRADE-3.x.md

```diff
@@ -32,6 +32,19 @@ extend the corresponding classes in `SonataAdminBundle\Action\`.
 If you need to style headers prefer to use CSS classes and not in the html DOM.
 In this case please use `header_class` option.
 
+## Deprecated returning other type than `ArrayCollection` from `SimplePager::getResults()`
+
+When calling `SimplePager::getResults()` on non-empty result which has set `$maxPerPage`, `ArrayCollection` would be returned instead of `array` as it is declared in `PagerInterface`. Update usage of `SimplePager::getResults()`, ensure you are transforming `Collection` to `array` and you aren't dealing with any of its methods.
+
+```
+// will return ArrayCollection on non-empty result and array on empty result
```

Comment:
Why an `ArrayCollection` and not just `Collection` or `iterable`?

Reply:
because only ArrayCollection will be returned

see
https://github.com/sonata-project/SonataAdminBundle/commit/1a40728ba8b7524ad34410ce217f236eb90af932#diff-7b1b93d6330bfae84c4609b520fc1a6aL80-L84

Reply:
This is better to work with interface. Maybe the code has to be updated in SimplePager

Reply:
Okay. This PR took here for 2y, what to close this gestalt :)